awx/main/models/events.py

```diff
@@ -619,14 +542,7 @@ def create_from_data(cls, **kwargs):
             kwargs.pop('created', None)
 
         sanitize_event_keys(kwargs, cls.VALID_KEYS)
-        kwargs.pop('workflow_job_id', None)
-        event = cls.objects.create(**kwargs)
-        if isinstance(event, AdHocCommandEvent):
-            analytics_logger.info(
-                'Event data saved.',
-                extra=dict(python_objects=dict(job_event=event))
-            )
-        return event
+        return kwargs
```

Comment:
If I was coming into the code base cold, this would make me a little confused about the naming and function of `create_from_data`

Reply:
Agreed - I changed this back so the method actually returns an object (even though it hasn't actually been saved/inserted yet).